src/main/kotlin/org/malachite/estella/queues/AdminQueue.kt

```diff
@@ -0,0 +1,36 @@
+package org.malachite.estella.queues
+
+import com.beust.klaxon.Klaxon
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory
+import org.springframework.amqp.rabbit.connection.ConnectionFactory
+import org.springframework.amqp.rabbit.core.RabbitAdmin
+import java.io.File
+import java.net.URI
+
+object AdminQueue {
+
+    private val env = System.getenv()
+
+    private val cloudamqpKey = "CLOUDAMQP_URL"
+    const val configPath = "config.json"
+
+    private fun getCloudAmqpFromConfig(): MutableMap<String, String> {
+        val file = File(configPath)
+        if (!file.exists()) throw NoSuchFileException(file)
+        return Klaxon().parse<MutableMap<String, String>>(file)!!
+    }
+
+    fun generateConnectionFactory(): ConnectionFactory {
```

Comment:
Potrzebujemy tego gdzieś poza tym obiektem? Ja bym to dal jako private i w ogóle w companion object żeby się to raz tylko zrobiło.

Reply:
Ogólnie  nie jestem przekonany, że wywoływanie za każdym razem nowego Connection factory jest dobrym pomysłem. Nie lepiej zrobić singleton?